packages/nvidia_nat_core/src/nat/data_models/atif/step.py

```
@@ -0,0 +1,130 @@
+
+    model_config = {"extra": "forbid"}
+
+    @field_validator("timestamp")
+    @classmethod
+    def validate_timestamp(cls, v: str | None) -> str | None:
+        """Validate that timestamp is a valid ISO 8601 string."""
+        if v is not None:
+            try:
+                datetime.fromisoformat(v.replace("Z", "+00:00"))
+            except ValueError as e:
+                raise ValueError(f"Invalid ISO 8601 timestamp: {e}") from e
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)